src/providers/secondary-media-loader.ts

```diff
@@ -0,0 +1,58 @@
+import ILoader = KalturaPlayerTypes.ILoader;
+
+const {RequestBuilder, ENUMS} = KalturaPlayer.providers;
+
+export class SecondaryMediaLoader implements ILoader {
+  _parentEntryId: string = '';
+  _requests: any[] = [];
+  _response: any = {};
+
+  static get id(): string {
+    return 'dualscreen';
+  }
+
+  /**
+   * @constructor
+   * @param {Object} params loader params
+   */
+  constructor(params: any) {
+    this._parentEntryId = params.parentEntryId;
+    const headers: Map<string, string> = new Map();
+    const request = new RequestBuilder(headers);
+    request.service = 'baseEntry';
+    request.action = 'list';
+    request.params = {
+      filter: {objectType: 'KalturaBaseEntryFilter', parentEntryIdEqual: this._parentEntryId},
+      responseProfile: {
+        type: ENUMS.RESPONSE_PROFILE_TYPE.INCLUDE_FIELDS,
+        fields: 'id'
+      }
+    };
+    this.requests.push(request);
+  }
```

Comment:
all of this is basically https://github.com/kaltura/playkit-js-providers/blob/master/src/k-provider/ovp/loaders/entry-list-loader.js with `parentEntryIdEqual`, no?
I suggest to see if we can reuse the loader and the data loader expects params so you can inject your `parentEntryIdEqual` as params.
This way you also "earn" the parsed objects.

Reply:
I know see we do not pass params object as is in the provider entry list loader, so what I wrote is not possible.

Reply:
this will require a bigger refactor in base providers - to allow passing the loader params as is and merge them with the base service params that we manage in core provider.
So for now I think we should keep with this, open for discussion between you @yairans :-)

Reply:
It is a bit complex. Today the entry-list-loader is expecting to have Base entry service list which is filtering differently the data (idEqual instead of parentEntryIdEqual). It is possible of course to re-use it but we need to find a way to override the filtering.
 What seems to me more simple and effective is to expose the response types. In dual screen I filtered only the Id in the response but if I would have gotten the entire response it would have simplify it.

Reply:
Just handled the response in the Setter